addons/html_editor/static/src/utils/dom_state.js

```diff
@@ -321,6 +321,26 @@ const priorityRestoreStateRules = [
         },
         { brVisibility: false, extraBRRemovalCondition: (brNode) => isFakeLineBreak(brNode) },
     ],
+    [
+        // Replace a space by &nbsp; when element last child is \uFEFF character.
+        {
+            direction: DIRECTIONS.LEFT,
+            cType1: CTYPES.CONTENT,
+            cType2: CTYPES.CONTENT,
+        },
+        {
+            lineBreakSpaceVisibility: (el) => {
+                if (
+                    el.nodeType === Node.ELEMENT_NODE &&
+                    !isBlock(el) &&
+                    el.hasChildNodes() &&
+                    el.lastChild.textContent.includes("\uFEFF")
+                ) {
+                    return true;
+                }
+            },
+        },
+    ],
```

Comment:
I just tested in jsfiddle and that space is visible even though the last child is a `\uFEFF`:
```html
<p contenteditable>a &#xFEFF</p>
```
What am I missing?

Reply:
I'm not sure I understand because in that case, since the test simply adds a BR, we don't need the FEFF, right? So that test should work like it always had up until now, shouldn't it?

Reply:
I have now added contentAfterEdit in the [test case](https://github.com/odoo/odoo/pull/179255/files#diff-9b188d1f8f7f065a6fb2a1e7375a20744e5a7d6f5596322aa7605d1de061fea4R197), which contains the `\ufeff` character. However, this character is removed during the cleanForSave process, so it does not appear in contentAfter.

Before my changes, only a `<br>` was added during the line break. As a result, the space was converted into `&nbsp;`, which prevented it from being removed during the cleanForSave process.

Reply:
I'm sorry, I don't understand. You say that before your changes the space was converted to a `&nbsp;`, but even with your changes the space is _still_ converted to a "&nbsp;" right? I mean, that's how it looks in the test case that you linked.

I tested in jsfiddle with a normal space and it looks to be visible, but only specifically because of the FEFF.
```js
<p><b>abc<br>&#xfeff</b> def</p></div>
```
If I remove the FEFF then it becomes invisible. Is that what you meant? That if you don't transform the space into a `&nbsp:`, even though it's not needed while editing, it's going to end up invisible when the FEFF are removed?

Reply:
I have applied the changes!